src/coreclr/gc/gcinterface.h

```diff
@@ -1066,12 +1066,18 @@ inline GC_ALLOC_FLAGS& operator&=(GC_ALLOC_FLAGS& a, GC_ALLOC_FLAGS b)
 #define UNCHECKED_OBJECTREF_TO_OBJECTREF(obj)       (obj)
 #endif
 
+#if !defined(TARGET_X86) || defined(TARGET_UNIX)
+#define USE_STACK_LIMIT
+#endif
+
 struct ScanContext
 {
     Thread* thread_under_crawl;
     int thread_number;
     int thread_count;
+#ifdef USE_STACK_LIMIT
```

Comment:
Is this a breaking change in the GC/EE contract?

cc @dotnet/gc

Reply:
I think so, thank you for spotting this. Let's keep this field in unconditionally.

Reply:
we really try to keep the #define's low in gcinterface.h because it'd require us to keep track of what version defines what. what would be the drawback to keep the code as is instead of adding this define?

Reply:
I can leave the code in and just let the limit to be zero on x86 windows. That should work.

Reply:
@Maoni0 I've modified the PR to minimalize the number of ifdefs overall, no more gcinterface.h changes anymore.

Reply:
awesome, thank you!